master/internal/api_tasks_intg_test.go

```diff
@@ -107,9 +113,96 @@ func TestPostTaskLogs(t *testing.T) {
 	}
 }
 
+func validateResponseSummary(t *testing.T, expectedTaskSummary *taskv1.AllocationSummary,
+	taskSummary *taskv1.AllocationSummary,
+) {
+	require.Equal(t, expectedTaskSummary.TaskId, taskSummary.TaskId)
+	require.Equal(t, expectedTaskSummary.AllocationId, taskSummary.AllocationId)
+	require.Equal(t, expectedTaskSummary.Name, taskSummary.Name)
+
+	require.Equal(t, len(expectedTaskSummary.Resources), len(taskSummary.Resources))
+	for i, r := range taskSummary.Resources {
```

Comment:
nit (non-blocking): If you know that every value in these structs is expected to be equal, you could use `reflect.DeepEqual(expectedTaskSummary.Resources, taskSummary.Resources)`

That being said, the way your doing it is perfectly fine and will give an easier to interpret error messages if the test fails